### ISSUE TITLE
Unable to build on Mac because the build can't find libavformat/avformat.h

### ISSUE THREAD

**Issue by bzbarsky** [servo/servo#12672]
I tried following the directions at https://github.com/servo/servo/blob/master/README.md to compile Servo on mac.  The build dies, like so:

     "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang" -cc1 -triple x86_64-apple-macosx10.10.0 -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -emit-obj -mrelax-all -disable-free -disable-llvm-verifier -main-file-name vmrs.c -mrelocation-model pic -pic-level 2 -mthread-model posix -mdisable-fp-elim -masm-verbose -munwind-tables -target-cpu core2 -target-linker-version 253.9 -v -gdwarf-2 -dwarf-column-info -ffunction-sections -fdata-sections -coverage-file /Users/bzbarsky/mozilla/servo/target/debug/build/video-metadata-68095025ad9189db/out/ffi/vmrs.o -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.0.2 -O0 -fdebug-compilation-dir /Users/bzbarsky/mozilla/servo/.cargo/git/checkouts/video-metadata-rs-58c16900f00fb0a1/master -ferror-limit 19 -fmessage-length 0 -stack-protector 1 -mstackrealign -fblocks -fobjc-runtime=macosx-10.10.0 -fencode-extended-block-signature -fmax-type-align=16 -fdiagnostics-show-option -o /Users/bzbarsky/mozilla/servo/target/debug/build/video-metadata-68095025ad9189db/out/ffi/vmrs.o -x c ffi/vmrs.c
    clang -cc1 version 7.0.2 based upon LLVM 3.7.0svn default target x86_64-apple-darwin14.5.0
    ignoring nonexistent directory "/usr/local/include"
    #include "..." search starts here:
    #include <...> search starts here:
     /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.0.2/include
     /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include
     /usr/include
     /System/Library/Frameworks (framework directory)
     /Library/Frameworks (framework directory)
    End of search list.
    In file included from ffi/vmrs.c:1:
    ffi/vmrs.h:4:10: fatal error: 'libavformat/avformat.h' file not found
    #include <libavformat/avformat.h>
             ^

I tried `port install ffmpeg` and `port install ffmpeg-devel`, both of which install the relevant header, but both put it in `/opt/local/include/libavformat/avformat.h` and the above header search path does not contain `/opt/local/include`.  Note the `ignoring nonexistent directory "/usr/local/include"` bit, in case it matters.

**Comment by clstl** [comment]
I had the same exact issue today. I have a "shady" workaround (works for now) , but not the solution
`ln -s /usr/local /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/local`

**Comment by GuillaumeGomez** [comment]
Look [here](https://github.com/servo/servo/issues/11882#issuecomment-236463715).

**Comment by larsbergstrom** [comment]
Yes, when I've seen this (on our OSX builders), we fixed it by doing:
`xcode-select --install`

Which I got from: http://stackoverflow.com/a/26404710

**Comment by bzbarsky** [comment]
`xcode-select --install` does not work, of course, because that doesn't change anything about `/opt/local/bin`.  When I run it, it (unsurprisingly) says:

> xcode-select: error: command line tools are already installed, use "Software Update" to install updates

because I've run it before already, obviously (can't really build Firefox otherwise!).

**Comment by m1c0l** [comment]
I'm getting a similar error on Linux when I run the nightly download:
```
./servo: error while loading shared libraries: libavformat.so.54: cannot open shared object file: No such file or directory

**Comment by birkoffe** [comment]
In windows nighly builds same error with missing avformat-57.dll and avutil-55.dll on servo start.

**Comment by clstl** [comment]
@larsbergstrom what version of OSX do you have there? I bet it is only El Capitan that does this. I had quite a few issues with paths over the past year.

**Comment by Ms2ger** [comment]
Should we turn ffmpeg off again?

**Comment by bzbarsky** [comment]
@clstl I have Yosemite.

**Comment by larsbergstrom** [comment]
That's really weird, because on my mac (10.11) and our builders (10.10), the files are in `/usr/local/include/libavformat`. The error that seems to generate this condition is that `/usr/local` gets set to no longer have read permission for the user (literally, goofball `d-wx--x--x`) permissions.

**Comment by ConnorGBrewster** [comment]
I had this issue earlier, `xcode-select --install` fixed it for me.

**Comment by bzbarsky** [comment]
@larsbergstrom Presumably both your mac and your builders are using homebrew.  I'm using macports.  The headers end up in a different place there: `/opt/local/include/libavformat`.

**Comment by larsbergstrom** [comment]
@bzbarsky Thanks for the clarification and I apologize for my assumption! Due, in part, to most CI systems that support OSX having moved over to HomeBrew, I'd forgotten that macports was still being supported and used :-)

We may have to consider adding a macOS builder that only has macports stuff installed and no homebrew if we want to continue supporting this developer configuration. Added an issue to our infra repo for discussion: https://github.com/servo/saltfs/issues/452

**Comment by jbriggs-zvelo** [comment]
I have a very similar problem on Fedora. My libavformat header files are installed in /usr/include/ffmpeg and so they aren't found by a simple include. They need -I/usr/include/ffmpeg set. The Rust build files  should be using pkg-config to locate these.

**Comment by GuillaumeGomez** [comment]
@jbriggs-zvelo: Look [here](https://github.com/servo/servo/issues/11882#issuecomment-236463715).

**Comment by bzbarsky** [comment]
Per irc discussion, using pkg-config would also work for macports, because then the right include path would get set.

**Comment by metajack** [comment]
@bzbarsky This should be fixed by #12683. I'll close once you confirm it's now working for you.

**Comment by bzbarsky** [comment]
Yep, confirmed that things build correctly now.  Thanks!
